File: Refrigtz/Syncronization.cs
using System;
using System.IO;
using System.Threading;
namespace Refrigtz
{
    [Serializable]
    class Syncronization
    {
        //Error Handling.
        static void Log(Exception ex)
        {
            try
            {
                Object a = new Object();
                lock (a)
                {
                    string stackTrace = ex.ToString();
                    File.AppendAllText(FormRefrigtz.Root + "\\ErrorProgramRun.txt", stackTrace + ": On" + DateTime.Now.ToString()); // path of file where stack trace will be stored.
                }
            }
            catch (Exception t) { Log(t); }
        }
        //
        bool LastState = false;
        public Syncronization(Thread t, int a)
        {
            try
            {
                if (a == 1)
                {
                    LastState = Mutex(t);
                }
                else if (a == 2)
                {
                    LastState = Event(t);
                }
                else if (a == 3)
                {
                    LastState = Semaphore(t);
                }
            }
            catch (Exception tt)
            {
                Log(tt);
            }
        }
        bool Mutex(Thread t)
        {
            try
            {
                //#pragma warning disable CS0618 // 'Thread.Suspend()' is obsolete: 'Thread.Suspend has been deprecated.  Please use other classes in System.Threading, such as Monitor, Mutex, Event, and Semaphore, to synchronize Threads or protect resources.  http://go.microsoft.com/fwlink/?linkid=14202'
                t.Suspend();
                //#pragma warning restore CS0618 // 'Thread.Suspend()' is obsolete: 'Thread.Suspend has been deprecated.  Please use other classes in System.Threading, such as Monitor, Mutex, Event, and Semaphore, to synchronize Threads or protect resources.  http://go.microsoft.com/fwlink/?linkid=14202'

            }
            catch (Exception tt)
            {
                Log(tt);
            }
            return true;
        }
        bool Event(Thread t)
        {
            try
            {
                t.Start();

            }
            catch (Exception tt)
            {
                Log(tt);
            }
            return true;
        }
        bool Semaphore(Thread t)
        {
            try
            {
                //#pragma warning disable CS0618 // 'Thread.Resume()' is obsolete: 'Thread.Resume has been deprecated.  Please use other classes in System.Threading, such as Monitor, Mutex, Event, and Semaphore, to synchronize Threads or protect resources.  http://go.microsoft.com/fwlink/?linkid=14202'
                t.Resume();
                //#pragma warning restore CS0618 // 'Thread.Resume()' is obsolete: 'Thread.Resume has been deprecated.  Please use other classes in System.Threading, such as Monitor, Mutex, Event, and Semaphore, to synchronize Threads or protect resources.  http://go.microsoft.com/fwlink/?linkid=14202'

            }
            catch (Exception tt)
            {
                Log(tt);
            }
            return true;
        }
    }
}
//End of Documents.

File: Refrigtz/Syncronization.vb
Imports System.IO
Imports System.Threading
Imports System.Text
Imports System.Linq
Imports System.Collections.Generic
Imports System
Namespace Refrigtz
    Class Syncronization
        'Error Handling.
        Shared Sub Log(ex As Exception)
            Try
                Dim stackTrace As String = ex.ToString()
                ' path of file where stack trace will be stored.
                File.AppendAllText("ErrorProgramRun.txt", stackTrace + ": On" + DateTime.Now.ToString())
            Catch t As Exception
                Log(t)
            End Try
        End Sub
        '
        Private LastState As Boolean = False
        Public Sub New()
        End Sub
        Public Sub Sync(t As Thread, a As Integer)
            Try
                If a = 1 Then
                    LastState = Mutex(t)
                ElseIf a = 2 Then
                    LastState = [Event](t)
                ElseIf a = 3 Then
                    LastState = Semaphore(t)
                End If
            Catch tt As Exception
                Log(tt)
            End Try
        End Sub
        Function Mutex(t As Thread) As Boolean
            Try

                t.Suspend()
            Catch tt As Exception
                Log(tt)
            End Try
            Return True
        End Function
        Function [Event](t As Thread) As Boolean
            Try

                t.Start()
            Catch tt As Exception
                Log(tt)
            End Try
            Return True
        End Function
        Function Semaphore(t As Thread) As Boolean
            Try

                t.[Resume]()
            Catch tt As Exception
                Log(tt)
            End Try
            Return True
        End Function
    End Class
End Namespace